shell/common/platform_util.h

```diff
@@ -41,7 +42,7 @@ void OpenExternal(const GURL& url,
                   OpenExternalCallback callback);
 
 // Move a file to trash.
-bool MoveItemToTrash(const base::FilePath& full_path);
+bool MoveItemToTrash(const base::FilePath& full_path, mate::Arguments* args);
```

Comment:
The function signature can be just `bool MoveItemToTrash(mate::Arguments* args)` and the `const base::FilePath& full_path` value can be still be checked for non-optionality with `args->Length()`.

Reply:
Putting the filrpath in the signature though ensures the error message for missing arguments is consistent across methods 

#justmy2cents

Reply:
In that case the implementation declaration should not be made aware of `mate::Arguments`, I would suggest doing the runtime parsing in `shell/common/api/*.cc` and call the implementation methods with right values.